tests/backend_test.py

```diff
@@ -1430,3 +1432,63 @@ def test_ibmq_local_emulator(
     counts = r.get_counts()
     # Most results should be (0,0) or (1,1):
     assert sum(c0 != c1 for c0, c1 in counts) < 25
+
+
+# https://github.com/CQCL/pytket-qiskit/issues/231
+def test_noiseless_density_matrix_simulation() -> None:
+    density_matrix_backend = AerDensityMatrixBackend()
+    assert density_matrix_backend.supports_density_matrix is True
+
+    assert isinstance(density_matrix_backend.backend_info.architecture, FullyConnected)
+
+    circ1 = Circuit(3).X(0).X(1).CCX(0, 1, 2)
+
+    output_state = np.array([0] * 7 + [1])
+
+    result1 = density_matrix_backend.run_circuit(circ1)
+
+    assert result1.get_density_matrix().shape == (8, 8)
+    assert np.allclose(
+        result1.get_density_matrix(), np.outer(output_state, output_state.conj())
+    )
+    # Example with resets and conditional gates
+    # Deterministic if input is a computational basis state
```

Comment:
Don't quite understand this comment. What exactly is deterministic?

Reply:
So despite the fact that the circuit contains measures and resets it prepares a state deterministically assuming that it starts from the $|0\rangle^{\otimes n}$ state.